syntax = "proto3";

option cc_generic_services = true;

package RPCTest;

message ResultCode {
    int32 errcode = 1; // 错误码
    bytes errmsg = 2;  // 错误信息
}

message RegisterRequest {
    bytes name = 1;
    bytes password = 2;
}

message RegisterResponse {
    ResultCode result = 1;
    bool success = 2;
}

message LoginRequest {
    bytes name = 1;
    bytes password = 2;
}

message LoginResponse {
    ResultCode result = 1;
    bool success = 2;
}

service UserServiceRpc {
    rpc Register (RegisterRequest) returns (RegisterResponse);
    rpc Login (LoginRequest) returns (LoginResponse);
}